test/basic/ecs_web_update/ecs_web_update_test.go

```diff
@@ -0,0 +1,236 @@
+	os.Setenv(terratest_aws.AuthAssumeRoleEnvVar, aptibleAccountRole)
+
+	vpcId := terraform.Output(t, terraformOptions, "vpc_id")
+	vpcAsset, vpcAws, vpcErr := getAptibleAndAWSVPCs(t, ctx, c, vpcId, "testecs-pub-to-priv-vpc")
+	if assert.NoError(t, vpcErr) {
+		assertCommonVpc(t, vpcId, vpcAsset, vpcAws)
+	}
+
+	ecsWebId := terraform.Output(t, terraformOptions, "ecs_web_id")
+	ecsWebAsset, ecsClusterAws, ecsServiceAws, ecsErr := getAptibleAndAWSECSServiceAndCluster(t, ctx, c, ecsWebId, "ecs-pub-web-test-web-cluster", "ecs-pub-web-test")
+	if assert.NoError(t, ecsErr) {
+		assertCommonEcs(t, ecsWebId, ecsWebAsset, ecsClusterAws, ecsServiceAws)
+	}
+
+	ecsLoadBalancerUrl := terraform.Output(t, terraformOptions, "loadbalancer_url")
+	ecsLbGet, ecsLbGetErr := insecureHttpClient().Get(fmt.Sprintf("https://%s", ecsLoadBalancerUrl))
+	if assert.NoError(t, ecsLbGetErr) {
+		assert.EqualValues(t, ecsLbGet.StatusCode, 200)
+	}
+
+	ecsServiceUrl := terraform.Output(t, terraformOptions, "web_url")
+	ecsUrlGet, ecsUrlGetErr := http.Get(fmt.Sprintf("https://%s", ecsServiceUrl))
+	if assert.NoError(t, ecsUrlGetErr) {
+		assert.EqualValues(t, ecsUrlGet.StatusCode, 200)
+	}
+
+	terraformRegistrySecretOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: "./secrets",
+		Vars: map[string]interface{}{
+			"organization_id":      os.Getenv("ORGANIZATION_ID"),
+			"environment_id":       os.Getenv("ENVIRONMENT_ID"),
+			"aptible_host":         os.Getenv("APTIBLE_HOST"),
+			"registry_secret_name": "registry1",
+			"registry_secret_value": map[string]string{
+				"username": os.Getenv("SECRET_REGISTRY_USERNAME"),
+				"password": os.Getenv("SECRET_REGISTRY_PASSWORD"),
+			},
+		},
+	})
+	terraform.InitAndApply(t, terraformRegistrySecretOptions)
+	defer terraform.Destroy(t, terraformRegistrySecretOptions)
+	registrySecretArn := terraform.Output(t, terraformRegistrySecretOptions, "registry_secret_arn")
+
+	terraformUpdateOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: ".",
+		Vars: map[string]interface{}{
+			"organization_id":          os.Getenv("ORGANIZATION_ID"),
+			"environment_id":           os.Getenv("ENVIRONMENT_ID"),
+			"aptible_host":             os.Getenv("APTIBLE_HOST"),
+			"dns_account_id":           os.Getenv("DNS_AWS_ACCOUNT_ID"),
+			"ecs_name":                 "ecs-pub-web-test",
+			"container_command":        []string{"nginx", "-g", "daemon off;"},
+			"container_image":          "ghcr.io/aptible/docker-hello-world-private:main",
+			"container_port":           80,
+			"container_name":           "nginx",
+			"is_public":                true,
+			"is_ecr_image":             false,
+			"vpc_name":                 "testecs-pub-to-priv-vpc",
+			"domain":                   "aptible-cloud-staging.com",
+			"subdomain":                "test-ecs-pub-integration",
+			"registry_credentials_arn": registrySecretArn,
+		},
+	})
+	defer cleanupAndAssert(t, terraformUpdateOptions)
```

Comment:
this is scary and i wouldn't do this. i think there's a few defers in here, there should only be one and i'd point it to the same set of options with a pointer if that's your desire (to have the final set), these will execute LIFO and it feels off

i'm down with doing this if you feel strongly but it feels weird

Reply:
I did remove this later

Reply:
But calling delete multiple times is fine, so I don't think this would cause issue?

Reply:
it should be idempotent, it just felt a bit off, as you could just reference a pointer that is mutating over the test suite and will pick up the latest vars